browser/resources/settings/brave_leo_assistant_page/memory_section.html

```diff
@@ -0,0 +1,302 @@
+    text-align: center;
+    font-weight: 600;
+  }
+
+  .empty-memories-desc {
+    color: var(--leo-color-text-tertiary);
+    text-align: center;
+    font-weight: 400;
+  }
+
+  cr-dialog [slot="title"] {
+    color: var(--leo-color-text-primary);
+    font: var(--leo-font-heading-h3);
+    display: flex;
+    padding: var(--leo-spacing-2xl);
+    align-items: flex-start;
+    gap: var(--leo-spacing-xl);
+    align-self: stretch;
+  }
+
+  cr-dialog [slot="body"] {
+    display: flex;
+    padding: var(--leo-spacing-m) var(--leo-spacing-2xl);
+    flex-direction: column;
+    align-items: center;
+    gap: var(--leo-spacing-2xl);
+    align-self: stretch;
+  }
+
+  cr-dialog [slot="button-container"] {
+    display: flex;
+    justify-content: flex-end;
+    align-items: flex-start;
+    gap: var(--leo-spacing-m);
+  }
+
+  /* Avoid stretching the buttons */
+  .action-box leo-button,
+  cr-dialog [slot="button-container"] leo-button {
+    width: auto !important;
+    min-width: unset !important;
+    flex: 0 0 auto !important;
+  }
+
+  leo-button.delete-all-button:not([isDisabled]) {
+    --leo-button-color: var(--leo-color-systemfeedback-error-text);
+    --primary-color: var(--leo-color-systemfeedback-error-background);
+    --bg: var(--leo-color-systemfeedback-error-background);
+  }
+
+  leo-button.delete-all-button[isDisabled] {
+    position: relative;
+    opacity: 0.65;
+    --leo-button-color: var(--leo-color-text-disabled);
+    border-radius: var(--leo-radius-full);
+  }
+
+  leo-button.delete-all-button[isDisabled]::before {
+    content: '';
+    position: absolute;
+    inset: 0;
+    background: var(--leo-color-icon-default);
+    opacity: 0.07;
+    border-radius: inherit;
+    pointer-events: none;
+    z-index: 0;
+  }
+
+  .dialog-delete-button {
+    --leo-button-color: var(--leo-color-systemfeedback-error-text);
+    --primary-color: var(--leo-color-systemfeedback-error-background);
+  }
+
+  .error-message {
+    display: flex;
+    align-items: center;
+    gap: var(--leo-spacing-s);
+    color: var(--leo-color-systemfeedback-error-text);
+    font: var(--leo-font-small-regular);
+  }
+  .error-message leo-icon {
+    color: var(--leo-color-systemfeedback-error-icon);
+  }
+</style>
+
+<div class="settings-box header">
+  <div class="flex">
+    <div class="main-header">$i18n{braveLeoAssistantYourMemoriesTitle}</div>
+    <div class="label secondary">
+      <span>$i18n{braveLeoAssistantYourMemoriesDesc} $i18nRaw{braveLeoAssistantMemoryLearnMore}</span>
+    </div>
+  </div>
+</div>
+
+<settings-toggle-button id="userMemoryEnabled"
+    icon="database"
+    pref="{{prefs.brave.ai_chat.user_memory_enabled}}"
+    label="$i18n{braveLeoAssistantUserMemoryEnabledLabel}">
+</settings-toggle-button>
+
+<div class="list-container">
+  <template
+    is="dom-if"
+    if="[[!hasMemories_(memoriesList_)]]"
+  >
+    <div class="empty-memories-table">
+      <div class="empty-memories-empty-state">
```

Comment:
do we need this div? seems like we could just use the parent element

Reply:
https://github.com/brave/brave-core/pull/30054/commits/8ed8849e997c58c40876c47ab2faa92f7c3ec3f2